domain/src/Aggregates/SharePoolingAsset/Entities/SharePoolingAssetAcquisition.php

```diff
@@ -74,14 +89,14 @@ public function increaseSameDayQuantity(Quantity $quantity): self
         $quantityToDeduct = Quantity::minimum($quantityToAdd, $this->thirtyDayQuantity);
         $this->thirtyDayQuantity = $this->thirtyDayQuantity->minus($quantityToDeduct);
 
-        return $this;
+        return $quantityToAdd;
```

Comment:
ℹ️ `increaseSameDayQuantityUpToAvailableQuantity` and `increaseThirtyDayQuantityUpToAvailableQuantity` (below) now return the quantity that has actually been applied, instead of duplicating that logic in `DisposalBuilder`